compiler/optimizer/src/main/java/edu/snu/onyx/compiler/optimizer/pass/compiletime/annotating/DefaultParallelismPass.java

```diff
@@ -107,4 +125,28 @@ static Integer recursivelySynchronizeO2OParallelism(final DAG<IRVertex, IREdge>
     }
     return myParallelism;
   }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final DefaultParallelismPass that = (DefaultParallelismPass) o;
+
+    if (desiredSourceParallelism != that.desiredSourceParallelism) {
+      return false;
+    }
+    return shuffleDecreaseFactor.equals(that.shuffleDecreaseFactor);
```

Comment:
Can we make the shuffleDecreaseFactor as an argument instead of a fixed value as well? (like the source parallelism)